spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/cache/CacheAutoConfigurationTests.java

```diff
@@ -97,6 +99,16 @@
 
 	private AnnotationConfigApplicationContext context;
 
+	@BeforeClass
+	public static void init() {
+		System.setProperty("hazelcast.jcache.provider.type", "server");
+	}
+
+	@AfterClass
+	public static void close() {
+		System.clearProperty("hazelcast.jcache.provider.type");
+	}
+
```

Comment:
I don't like that. You shouldn't mess with this in the general `CacheAutoConfiguration`. Is it because the client is now on the classpath and the JCache support doesn't know what to do? 

If that's the case, we should fix the JCache implementation to set the right flag based on the `HazelcastInstance` that is available.